pyasice/xmlsig.py

```diff
@@ -330,12 +330,25 @@ def set_root_ca_cert(self, root_cert: Union[Certificate, bytes]) -> "XmlSignatur
 
         :param root_cert: can be a PEM- or DER-encoded bytes content, or an `oscrypto.Certificate` object
         """
+        self.add_cert(root_cert, {"Id": f"{self.NEW_SIGNATURE_ID}-ROOT-CA-CERT"})
+        return self
+
+    def add_cert(self, cert: Union[Certificate, bytes], attrib: Optional[dict]) -> "XmlSignature":
+        """Add a cert. Latvian EDOC must have all of certs used in the xml (Root, OCSP and TimeStamp)
+           This is mandatory for Latvian EDOC format
+
+        :param cert: can be a PEM- or DER-encoded bytes content, or an `oscrypto.Certificate` object
+        :param attrib: dict with attributes for <EncapsulatedX509Certificate> tag
+        """
         certs_node = self._get_node("xades:CertificateValues")
         ca_node = etree.Element("{%s}EncapsulatedX509Certificate" % self.NAMESPACES["xades"])
-        ca_node.attrib["Id"] = "%s-ROOT-CA-CERT" % self.NEW_SIGNATURE_ID
-        if not isinstance(root_cert, Certificate):
-            root_cert = load_certificate(root_cert)
-        ca_node.text = base64.b64encode(root_cert.asn1.dump())
+
+        for name, value in attrib.items():
```

Comment:
The type hints state that `attrib` is optional while the code expects them to always be set. Please wrap this for loop into `if attrib is not None:`

Reply:
Sorry, it's my fault, you definitely right. I suggest to set the default value of `attribs` params to {} imho it more clean solution. Look at new PR.

Reply:
Do not use `{}` as default argument. Wrap the for-loop or add snippet below.

```
if attrib is None:
    attrib = {}
```

Reply:
Ok, guys, you are talking about [Mutable defaults arguments](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) ? Therefore `{}` as a default parameter is bad style, isn't it?

What do you think about to change parameter to `attribs: Optional[Dict] = None` and wrapping loop like @Jyrno42 advised ?

```
if attribs is not None: 
    for name, value in attribs.items(): 
          ca_node.attrib[name] = value  
```

Reply:
Yeah that should work

Reply:
> Ok, guys, you are talking about Mutable defaults arguments ? Therefore {} as a default parameter is bad style, isn't it?

Exactly

Reply:
Thanks guys for cooperation. It was a good experience for me.